gateway/src/main/java/org/georchestra/gateway/security/oauth2/LdapEnabledCondition.java

```diff
@@ -0,0 +1,25 @@
+package org.georchestra.gateway.security.oauth2;
+
+import com.jayway.jsonpath.JsonPath;
+import org.springframework.beans.factory.config.YamlMapFactoryBean;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+import java.util.LinkedHashMap;
+
+public class LdapEnabledCondition implements Condition {
+
+    @Override
+    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+        Resource resource = new FileSystemResource(
+                System.getProperty("georchestra.datadir", "/etc/georchestra") + "/gateway/security.yaml");
+        YamlMapFactoryBean yaml = new YamlMapFactoryBean();
+        yaml.setResources(resource);
+        LinkedHashMap<String, LinkedHashMap<String, Object>> ldap = JsonPath.read(yaml.getObject().get("georchestra"),
+                "$.gateway.security.ldap");
+        return ldap.values().stream().anyMatch(el -> el.containsKey("enabled") && ((Boolean) el.get("enabled")));
+    }
```

Comment:
I think the `@Value` annotation is meant for this, I find very strange the need to read config files by ourselves

Reply:
@value testé, non fonctionnelle. ça retourne null!

Reply:
@pmauduit , je pense que c'est le meme probleme, les conditions s'executent avant la lecture de properties

Reply:
There definitely must be another way than reading config files